django/db/models/fields/__init__.py

```diff
@@ -2380,13 +2380,19 @@ def get_internal_type(self):
         return "UUIDField"
 
     def get_db_prep_value(self, value, connection, prepared=False):
-        if isinstance(value, six.string_types):
-            value = uuid.UUID(value)
-        if isinstance(value, uuid.UUID):
-            if connection.features.has_native_uuid_field:
-                return value
-            return value.hex
-        return value
+        if value is None:
+            return None
+        if not isinstance(value, uuid.UUID):
+            try:
+                value = uuid.UUID(value)
+            except (ValueError, TypeError):
+                raise
+            except:
```

Comment:
UUID throws `AttributeError` when you do `uuid.UUID({})`, so that. I hadn't thought of `KeyboardInterrupt` though. I'll change it to `except AttributeError`.